.circleci/config.yml

```diff
@@ -98,10 +98,9 @@ jobs:
       - image: pafortin/goaws
       - image: cimg/mysql:5.7
       - image: jdlk7/firestore-emulator
-        environment:
-          NODE_ENV: dev
-          AUTH_FIRESTORE_EMULATOR_HOST: localhost:9090
-
+    environment:
+      NODE_ENV: dev
+      FIRESTORE_EMULATOR_HOST: localhost:9090
```

Comment:
> It was affecting CI. It was not a problem in the past because
> * tests didn't eagerly use Firestore; that changed with my change to load the docs on the config manager's instantiation

If we weren't using the emulator in the past, then were we using a real Firestore instance?

I see the default `keyFileName` in config/index.ts, but does that path point to a real file with credentials in our CI context? How can you tell?

I don't see how the PaymentConfigManager tests could pass without either the emulator running or a real Firestore instance.

> * a hard coded but rather incomplete mock, which I deleted in this PR

Where was this mocked? I'm looking at the PaymentConfigManager's test, and previously it was requiring the module and I don't see that `load` was mocked anywhere before it was called.

Reply:
> > It was affecting CI. It was not a problem in the past because
> > 
> > * tests didn't eagerly use Firestore; that changed with my change to load the docs on the config manager's instantiation
> 
> If we weren't using the emulator in the past, then were we using a real Firestore instance?
> 
> I see the default `keyFileName` in config/index.ts, but does that path point to a real file with credentials in our CI context? How can you tell?
> 

If you look at `firestore-db.ts` you linked above you see that it only check for the config value existence, not the file at the path of the config value.

> I don't see how the PaymentConfigManager tests could pass without either the emulator running or a real Firestore instance.
> 

It depends on the test.  auth server has multiple sets of tests.  Some were using the emulator.  Some failed GCP auth because they weren't.  And some failed because the mock (see blow) just returns an empty object on `.collection()`.

> > * a hard coded but rather incomplete mock, which I deleted in this PR
> 
> Where was this mocked? I'm looking at the PaymentConfigManager's test, and previously it was requiring the module and I don't see that `load` was mocked anywhere before it was called.

Here is the mock: https://github.com/mozilla/fxa/pull/12196/files#diff-2ee317a43f8c0b03915fdd7e636d6287de7507c8f6635063629a1f32f828a925L30-L32


The output at https://app.circleci.com/pipelines/github/mozilla/fxa/28648/workflows/25eee859-a3b1-4df8-b113-ce230bf85c10/jobs/317859/steps shows the multiple ways the tests could failed.  `Error: 16 UNAUTHENTICATED: Failed to retrieve auth metadata with error: key must be a string, a buffer or an object` when they are not using the emulator.  `TypeError: this.productConfigDbRef.select is not a function` due to `collection` in the mock returning an empty object (`this.productConfigDbRef` is that empty object in the test).